src/lib_python.rs

```diff
@@ -0,0 +1,105 @@
+use std::path::Path;
+
+use anyhow::Result;
+use pyo3::exceptions::PyException;
+use pyo3::prelude::*;
+use pyo3::types::PyString;
+use rustpython_parser::lexer::LexResult;
+
+use crate::checks::{Check, CheckCode};
+use crate::directives;
+use crate::linter::check_path;
+use crate::rustpython_helpers::tokenize;
+use crate::settings::configuration::Configuration;
+use crate::settings::{flags, Settings};
+use crate::source_code_locator::SourceCodeLocator;
+
+#[pyclass]
+#[derive(Clone)]
+struct Location {
+    #[pyo3(get)]
+    row: usize,
+    #[pyo3(get)]
+    column: usize,
+}
+
+#[pyclass]
+struct Message {
+    #[pyo3(get)]
+    code: CheckCode,
+    #[pyo3(get)]
+    message: String,
+    #[pyo3(get)]
+    location: Location,
+    #[pyo3(get)]
+    end_location: Location,
+    // TODO(rgerecke): Include fix
+}
+
+// Using `#[pyclass]` on the `CheckCode` enum is incompatible with serde,
+// because this generates unsafe code.
+// TODO(rgerecke): Perhaps we want to generate module-level constants instead?
+impl IntoPy<PyObject> for CheckCode {
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        PyString::new(py, self.as_ref()).into()
+    }
+}
+
+fn inner_check(contents: &str) -> Result<Vec<Check>> {
+    let configuration = Configuration::default();
+    let settings = Settings::from_configuration(configuration, Path::new("."))?;
+
+    // Tokenize once.
+    let tokens: Vec<LexResult> = tokenize(contents);
+
+    // Initialize the SourceCodeLocator (which computes offsets lazily).
+    let locator = SourceCodeLocator::new(contents);
+
+    // Extract the `# noqa` and `# isort: skip` directives from the source.
+    let directives = directives::extract_directives(&tokens, &locator, directives::Flags::empty());
+
+    // Generate checks.
+    let checks = check_path(
+        Path::new("<filename>"),
+        None,
+        contents,
+        tokens,
+        &locator,
+        &directives,
+        &settings,
+        flags::Autofix::Enabled,
+        flags::Noqa::Enabled,
+    )?;
+
+    Ok(checks)
+}
+
+#[pyfunction]
+fn check(contents: &str) -> PyResult<Vec<Message>> {
+    // TODO(rgerecke): Accept settings
```

Comment:
We should also accept an optional filename.

Reply:
On that topic, do we mirror the current API 1:1, i.e. should this also resolve settings by looking into the filesystem?

Reply:
Yeah I think we should. Similar to the Rust API (which doesn't accept settings... but should).

I think the rules are:

- If settings are provided, use those.
- Else, if filepath is provided, infer.
- Else, use defaults.